### ISSUE TITLE
Container and codec limitation notification

### ISSUE THREAD

**Issue by tin2tin** [elmopl/ktba#8]
I noticed that Parallel Render will only work with the mpeg-4 container and a h.264 codec when using Ffmpeg video. I guess, that this is not a bug, but probably a limitation. Apparently, ffmpeg does not have a codec limitation when concatenating media files. Is this an extension issue? Anyway, not having the right container and codec will cause errors, so maybe a check is needed, if these settings are correct, and inform the user if they are not correct.

**Comment by elmopl** [comment]
I tried with:

![image](https://user-images.githubusercontent.com/285350/44953977-761f7a00-ae93-11e8-8c07-954f841194b6.png)

And it seems to have worked fine.
I probably should check if output is video, but other than that I can't seem to reproduce any issues.

What settings exactly did you try to use? 
What was ffmpeg error?

**Comment by elmopl** [comment]
I have added fix to hide `parallel render` from available operators when render file format is wrong.
I am not convinced that is the best UX for this. Maybe it should always be there and when invoked with incorrect values it should fail with some error before it starts to render?
Here my issue I think is what is usual Blender UI/flow to indicate mismatch in configuration like this.

**Comment by elmopl** [comment]
So I tried:
 - Quicktime causes my Blender to crashes Blender
 - DNxHD doesn't work even without mu plug-in (with the error about mismatch in resolution)

I am not quite sure how I could discover those things are not going to work when even Blender itself doesn't seem to handle those cases (i.e. I can select those options, but they will not render even without my add-on).

I think you should first try to see if with current render config you can just "render animation" without using my add-on (even if you just try to render 10-20 frames only). If Blender can't render it, then obviously using my add-on will not help.

So in short: I don't think those two above issue are problems with add-on itself as it seems that problem is there with just pure Blender.

**Comment by elmopl** [comment]
![image](https://user-images.githubusercontent.com/285350/44960580-f973b680-aef9-11e8-8673-3956df198bdf.png)

I moved configuration to a section under Render.
There is also an entry in "Render" menu.

I will work a bit more later on the popup you get just before you start parallel render.

**Comment by tin2tin** [comment]
Wow, that's super cool with the UI! AVI RAW and Quicktime works here now. :-) DNxHD codec is not working here in Blender natively, so you're right that must be a bug in Blender.

Unfortunately, I'm still getting the error with a fresh and clean start up of Blender with an unsaved .blend.  I had a look at the Power Sequencer add-on. And they are forcing the user to save the project this way:

> if not bpy.data.is_saved:
            layout.label('Please save your project')
            layout.operator(
            'wm.save_as_mainfile', icon='SAVE_AS', text='Save as')

From here: https://github.com/GDquest/Blender-power-sequencer/blob/master/ui/power_sequencer_menu.py

**Comment by elmopl** [comment]
For the unsaved .blend file there is #7 (it is easier to keep track of issues if they are separate).

> A thought on bad codecs/containers, maybe the addon could render one frame of the video and if it fails, it will abort the entire operation and give the user a message of "Unsupported codec/container"?

That did cross my mind and currently am contemplating if that's the best I can do.
It might be what I will end up doing.

BTW. interesting add-on. Haven't seen it before and seems like it might be useful.

**Comment by tin2tin** [comment]
It seems to me that it is now only bad containers/codecs causing errors when rendering(outside the not-saved-.blend-error), so maybe a simple way to get around this problem is to add a line of text in the progress bar in the header after the error message, something along the lines of "Try a different container or codec.", and this may also work and be valid for image formats, so maybe no more need to close down the UI when ex. PNG?

**Comment by Coarist** [comment]
Blender 2.81a VSE Rarallel Render, "Encoding - Container"

    Matroska: crash Blender during final "mixing down audio" stage.
    MPEG-4: Good, as reported in tine2tin's posting above.. 

Initially Blender crashed and dismissed itself after parallel render has almost finish showing "mixing down audio". Having read this posting, I changed "Encoding - Container" from "Matroska" to "MPEG-4". Parallel VSE render completed successfully after this change. Settings that works:

    Blender 2.81a version, Windows 10 x64
    FFmpeg video (Output File Format)
    MPEG-4 (Encoding-Container)
    H.264 (Video Codec)
    MP3 (Audio Codec)

Parallel Render config:

    2 Blender Instances
    4 Number of Parts

Results 30FPS, 3742 frames, 1920x1080 HD

    15 minutes, about 43% CPU utilisation, standard Blender VSE
    10 minutes, about 70% CPU utilisation, Blender VSE with Parallel Render

GPU utilisation also increase correspondingly when using Parallel Render. 

A big thank you.
